### PR TITLE
schema for pdos

### PR DESCRIPTION
**Pull request by unkcpz**
Schema part separated from https://github.com/osscar-org/widget-bandsplot/pull/15 for easy review.

### CODE REVIEW

**Review by dou-du — APPROVED** [review]

**Comment by unkcpz** [comment]
Hi @dou-du, I'd appreciate if you can have a look at the schema and at least let me know if you are okay with it. Thanks.

File: widget_bandsplot/schemas/pdos.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                    "x": {
+                        "type": "array",
+                        "items": {"type": "number"}
+                    },
+                    "y": {
+                        "type": "array",
+                        "items": {"type": "number"}
+                    },
+                    "lineStyle": {"enum": ["solid", "dash"]},
+                    "borderColor": {"type": "string"},
```
Comment (dou-du) [inline widget_bandsplot/schemas/pdos.json:25]: It is possible to verify whether it is a valid color string?

**Comment by yakutovicha** [comment]
Hey @unkcpz, I believe there are still quite a number of file updates related to the pre-commit. Also, not sure why ci.yml is kept here. I believe it should remain in #15.

File: widget_bandsplot/schemas/pdos.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    "backgroundColor": {"type": "string"}
```
[the remark this answers is not in the document]
Comment (dou-du) [reply widget_bandsplot/schemas/pdos.json:26]: > Should we also add transparency for the background as an additional parameter?

@unkcpz asked the same for the transparency. We can have the "alpha" value.

File: widget_bandsplot/schemas/pdos.json (hunk 1 above)
Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:25]: This depends on which values are allowed, I think currently in the javascript, the hex is used for setting the color.
If this is the case, we can have the schema for RGB color as: https://github.com/the-grid/apidocs/issues/26 

```json
"hexcolor": {
  "description": "RGB color",
  "format": "rgbcolor",
  "pattern": "^#([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$",
  "example": "#aa33cc",
  "type": "string"
},
```

**Comment by unkcpz** [comment]
> Hey @unkcpz, I believe there are still quite a number of file updates related to the pre-commit. Also, not sure why ci.yml is kept here. I believe it should remain in https://github.com/osscar-org/widget-bandsplot/pull/15.

The `ci.yml` for this PR doesn't have pre-commit check but runs the test of the schema. #15 will only contain the pre-commit check in the end.

File: widget_bandsplot/schemas/pdos.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:26]: @yakutovicha Thanks for the review! 
That's true there are more things to add to the schema. But the purpose of this PR is to first time introduce the schema to the repository and we (me and Dou mostly) have a better way to make changes. Adding new field to the schema requires also adapt javascript to working with it. 

The way I proposed is in the future if we gonna add more features, always start by changing from schema and implementing the feature to the javascript at the same time.

File: widget_bandsplot/schemas/pdos.json (hunk 1 above)
Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:25]: I was asking what values are supported by the library you used in JavaScript, how many colors name are supported there?

Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:25]: We are going to support only common colors like red, blue, black, green or more such as list here https://matplotlib.org/stable/gallery/color/named_colors.html?

**Review by yakutovicha — COMMENTED** [review]
Alright, for me it looks good to go. Looking at the code I would suggest modernising the `setup.py`. Maybe use https://github.com/aiidalab/aiidalab for an example (see the `setup.cfg` and `setup.py`)

We don't need `requirements.txt` and `requirements_test.txt`. It all could be incorporated into one `setup.cfg` file.

File: widget_bandsplot/schemas/pdos.json (hunk 1 above)
Comment (dou-du) [reply widget_bandsplot/schemas/pdos.json:25]: "When supplying colors to Chart options, you can use a number of formats. You can specify the color as a string in hexadecimal, RGB, or HSL notations. "

Comment (dou-du) [reply widget_bandsplot/schemas/pdos.json:25]: https://www.w3schools.com/colors/colors_names.asp

Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:25]: I see, then I'd say let's keep the string without any pattern, it's not easy to support all of those explicitly.

File: widget_bandsplot/schemas/pdos.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (yakutovicha) [reply widget_bandsplot/schemas/pdos.json:26]: Alright, good to know! Thank you for the explanation @unkcpz!

That basically means in the current PR you are trying to reflect the *old" schema, that is about to be updated, right?

[the remark this answers is not in the document]
Comment (unkcpz) [reply widget_bandsplot/schemas/pdos.json:26]: Yes, correct ;) only separate the PR for easy review.

**Pull request merged**